### ISSUE TITLE
Add support for creating profiles and associate rules with AAP profile

### ISSUE THREAD

**Issue by ganeshrn** [ansible/ansible-lint#2109]
##### Summary

- [ ] Add support for creating custom profiles and associate rules with the given  profile. For example `aap` (Ansible automation platform) can be a profile and the rules associated with this profile indicates the good practise to be followed while developing content for AAP.

- [ ] Allow customised error levels based on profile

##### Issue Type

- Feature request

**Comment by ssbarnea** [comment]
Some ideas about how I would see profiles configurable:
```yml
profiles:
  default:  # <-- that is the implicit profile
    description: ...
    rules:
      risky-file-permissions: null # <-- to disable/neuter a specific rule
      yaml:
        severity: 0.2 # <-- to override default rule severity
      xxx: ... # we will ignore any unrecognized rule-id
  aap:
    description: ...
    rules:
       builtins-only:
         severity: 1.0 # make it critical
```

@cidrblock Any other ideas?

**Comment by cognifloyd** [comment]
I've talked with @bcoca about this before. Here's another use case up keep in mind for profiles:
Ansible upgrades.

An upgrade between Ansible versions is currently documented in an upgrade guide. With the `--write` feature, rules will gain the ability to fix issues and not just identify them. So, we can have a profile for each upgrade guide to facilitate doing much of the grunt work, upgrading content between Ansible versions.

If the profile has to explicitly list roles to "neuter" or opt-out of, then that will make developing more rules problematic, because you'll have to exclude those new rules from older profiles.

Maybe the profiles need the option to exclude all rules except those listed in the profile definition.

**Comment by cidrblock** [comment]
based on our chat:
```
---
# defaults:
#  enabled: True
#  severity: high ?

# required:
#   description (for both profile and rule)


# We need to error out:
# if a rule is not found
# if duplicate profile name exist

# ansible-lint --profile ansible-lint.aap
# ansible-lint --profile company.profile (searches in the users ansible-lint.cfg)
# ansible-lint --settings

# long-term goals
# - pull profile using URL
# - package profiles as python packages

# these are defined internal to ansible-lint
---
profiles:
  ansible-lint.base:
    description: The base profile, only the specified rules are used
    rules:
      rule_a: #this is the primary key
        severity: high
        description: A description
        enabled: True # True is the default
      rule_b:
        severity: medium
        description: Another description
      rule_c:
        severity: low
        description: A third description

  ansible-lint.default: 
    description: The lint default profile
    extends: ansible-lint.base
   
  ansible-lint.profile_1: 
    description: A profile with a different set of rules
    extends: ansible-lint.base
    rules:
      rule_a:
        severity: medium
        description: A description
      rule_b:
        severity: medium
        description: Another description
        enabled: False # << disables the rule from base


---
# these are defined in the user ansible-lint settings file
# a user defined profile cannot start with 'ansible-lint'
---
profile: company_a.base # this denotes which profile is in use, overridden with CLI

profiles:
  user.base: #shown here in the same file, but should be it's own file
    description: A profile with a different set of rules
    extends: ansible-lint.base
    rules:
      rule_b:
        severity: medium
        description: Another description
        enabled: False # << disables the rule from base
 
  company_a.base:
    extends: ansible-lint.base


  company_a.strict:
    extends: company_a.base
    # turn on more rules

  company_a.loose:
    extends: company_a.base
    # disable many rules

  company_b.ci:
    extends: ansible-lint.base
    rules:
      # turn on all rules
      # set many to warning

  company_b.dev:
    extends: ansible-lint.base

```
